Detectors/FOCAL/simulation/src/Detector.cxx

```diff
@@ -453,7 +471,14 @@ void Detector::ConstructGeometry()
 
   TVirtualMC::GetMC()->Gsvolu("FOCAL", "BOX", getMediumID(ID_AIR), pars, 4);
   mSensitiveHCAL.push_back("FOCAL");
+  mSensitiveECALPad.push_back("FOCAL");
+  mSensitiveECALPix.push_back("FOCAL");
```

Comment:
I think it is enough to have one container for sensitive volumes. The idea behind it is to declare all volumes as sensitive in the framework (so the FairRoot step manager calls ProcessHits for every step allowing us to build the decay chain). Separate lists would only make sense to distinguish in the second step of ProcessHits the "real" sensitive volumes for calling the proper ProcessHits function of the 3 subsystems.

Like this you declare the top volume 3 times.

Reply:
Hi Markus, Do you suggest to create one container for the purpose of geometry creation and separate containers to keep ECAL Pads or Pixels or HCAL for the step manager?

Reply:
Indeed this is the idea. The reason is that we need to call ProcessHits for every step to build the tree of decay chains, as we cannot access them by the stack, and if we would do this only for the real sensitive volumes we would get broken chains - we saw this in EMCAL. For this purpose we don't distinguish between the subdetectors and can use a common list. Once this is done we do the separate volume check ourselves and redirect to separate ProcessHits functions for Pads, Pixels and HCAL based on the real sensitive volume ID.